src/mediaQueryListener.js

```diff
@@ -0,0 +1,13 @@
+const mediaQueryListener = (mediaQueries, update) => {
+  const mqlList = Object.keys(mediaQueries).map(size => {
+    const mql = window.matchMedia(mediaQueries[size])
+    mql.addListener(update)
+    return mql
+  })
+
+  return () => {
```

Comment:
in terms of coding style I'd prefer `() => mqlList.forEach((mql) => mql.removeListener(update))`